### ISSUE TITLE
Support of item creation order manipulation in groups

### ISSUE THREAD

**Issue by tatatupi** [BlueBrain/HighFive#485]
The original HDF5 API supports changing the item creation order:

```c++
hid_t group_creation_plist;
group_creation_plist = H5Pcreate(H5P_GROUP_CREATE);
H5Pset_link_creation_order(group_creation_plist, H5P_CRT_ORDER_TRACKED | H5P_CRT_ORDER_INDEXED);
```

I couldn't find a way to do it for the HighFive API however. I noticed some creation properties for datasets. Should there be creation properties for groups as well?

**Comment by tdegeus** [comment]
As far as I know it is not possible, but this might be a statement out of ignorance.

**Comment by tatatupi** [comment]
> As far as I know it is not possible, but this might be a statement out of ignorance.

I believe you are right. I was testing these HDF5 creation orders as well, and they did not work as expected. I will be ordering my data externally. No problem. 

Thanks though!

**Comment by alkino** [comment]
We have a PR that can be let you do what you want: #683 
Can you elaborate when you said it was not working `as expected`?

**Comment by alkino** [comment]
Fixed by #683
